xmodule/video_block/video_handlers.py

```diff
@@ -495,118 +495,179 @@ def studio_transcript(self, request, dispatch):
+                        metadata={
+                            'file_format': Transcript.SJSON,
+                            'language_code': new_language_code
+                        },
+                        file_data=ContentFile(sjson_subs),
+                    )
 
-                return Response(status=200)
+                # If a new transcript is added, then both new_language_code and
+                # language_code fields will have the same value.
+                if language_code != new_language_code:
+                    self.transcripts.pop(language_code, None)
+                self.transcripts[new_language_code] = filename
 
-            elif request.method == 'GET':
-                language = request.GET.get('language_code')
-                if not language:
-                    return Response(json={'error': _('Language is required.')}, status=400)
+                if isLibrary:
+                    self._save_transcript_field()
+                response = Response(json.dumps(payload), status=201)
+            except (TranscriptsGenerationException, UnicodeDecodeError):
+                response = Response(
+                    json={
+                        'error': _(
+                            'There is a problem with this transcript file. Try to upload a different file.'
+                        )
+                    },
+                    status=400
+                )
+        return response
 
-                try:
-                    transcript_content, transcript_name, mime_type = get_transcript(
-                        video=self, lang=language, output_format=Transcript.SRT
-                    )
-                    response = Response(transcript_content, headerlist=[
-                        (
-                            'Content-Disposition',
-                            f'attachment; filename="{transcript_name}"'
-                        ),
-                        ('Content-Language', language),
-                        ('Content-Type', mime_type)
-                    ])
-                except (UnicodeDecodeError, TranscriptsGenerationException, NotFoundError):
-                    response = Response(status=404)
+    def _studio_transcript_delete(self, request):
+        """
+        Delete transcript. Used in "DELETE" method in `studio_transcript`
+        """
+        request_data = request.json
+
+        if 'lang' not in request_data or 'edx_video_id' not in request_data:
+            return Response(status=400)
+
+        language = request_data['lang']
+        edx_video_id = clean_video_id(request_data['edx_video_id'])
 
+        if edx_video_id:
+            delete_video_transcript(video_id=edx_video_id, language_code=language)
+
+        if isinstance(self.usage_key.context_key, LibraryLocatorV2):
+            transcript_name = self.transcripts.pop(language, None)
+            if transcript_name:
+                lib_api.delete_library_block_static_asset_file(
```

Comment:
Note (please add as a comment): In the future, we need a proper XBlock API like `self.static_assets.delete(...)` instead of coding these runtime-specific/library-specific APIs.

Reply:
Added https://github.com/openedx/edx-platform/pull/36058/commits/6903be2b674410d06f07572f7acc6c37b3f23482